### DISCUSSION TITLE
Incomplete API Docs (trait implementations)

### DISCUSSION THREAD

**Discussion by b-naber** [actix/actix-web#2475] [General]
This is more of an issue, but since the issues section explicitly asks for bugs, I decided to post this here.

I noticed that the API docs are somewhat confusing in that it's not explicit that some types implement certain traits that are required in the public API. E.g. `HttpServer` requires an `impl IntoServiceFactory` to be returned in the closure parameter. `App` does implement `IntoServiceFactory` but this information is nowhere to be found in the docs. There are other such occurrences where the interdependence between the actix crates results in similar problems. Are there plans to improve this?

**Comment by robjtede** [comment]
I guess the answer to this is: we would appreciate specific examples (in issues would be good) of all cases where documentation is confusing to the folks who do not work on it day-to-day. `HttpServer`'s `IntoServiceFactory` bound is a good example and can certainly be improved in the documentation.

There are still certain limitations of Rust's type system that make this sort of indirection necessary and helping newcomers understand this hurdle is definitely a goal of Actix Web.
